src/pool/srv_pool.c

```diff
@@ -925,14 +928,52 @@ handle_event(struct pool_svc *svc, struct pool_svc_event *event)
 		goto out_prop;
 	}
 
-	rc = pool_svc_exclude_rank(svc, event->psv_rank);
-	if (rc != 0) {
-		D_ERROR(DF_UUID": failed to exclude rank %u: "DF_RC"\n", DP_UUID(svc->ps_uuid),
-			event->psv_rank, DP_RC(rc));
-		goto out_prop;
-	}
+	if (DAOS_POOL_SUPPORT_REINT_ONLINE == 0)
+		goto DEAD;	
 
-	D_DEBUG(DB_MD, DF_UUID": excluded rank %u\n", DP_UUID(svc->ps_uuid), event->psv_rank);
+	if (event->psv_type == CRT_EVT_ALIVE) {
+        dom = pool_map_find_node_by_rank(svc->ps_pool->sp_map, event->psv_rank);
```

Comment:
I wonder if there is an easy way to figure out the local rank status, instead of always checking pool map, since there might be a lot alive ping.

Reply:
Hmm, I thought before reintegration, there are some preparation work from mgmt, like ds_mgmt_pool_target_update_state(),

How is this auto swim alive notification sync with this? 

@liw  did we discuss any protocols about the automatic reintegration process?

Reply:
@wangdi, I don't think reint by swimming here should do the same preparation work like ds_mgmt_pool_target_update_state, because if it is new to be added, it can't be found in poolmap

Reply:
as for syncing with reintegration by command, there is a ABT_rwlock, which can guarantee the correctness